### DISCUSSION TITLE
Intellij-micropython not working

### DISCUSSION THREAD

**Discussion by IRBaboon595** [JetBrains/intellij-micropython#285] [General]
Lately I succesfully worked with Pycharm and Micropython. But someafter 2023 and 2024.1 (for now) the toolchain stoped work completely. 
For now I have Pycharm 2024.1; Micropython-plugin for PyCharm 1.4.0; Micropython firmware for Raspberry Pi Pico 1.22.2.
For what I see when I start to flash the MC - it stops on soft reboot but the REPL do not start at all.
And I faced this problem in the past after I last worked with PiPico in 2023.
Could you please clarify the problem for me and maybe make a fix?
Thanks in advance!

**Comment by philliphodges** [comment]
Same here.. The plugin is no longer working.. REPL fails to open/connect.. Plus only one file is pushed to the device.. It should be possible to push multiple files.. Such a shame as the only tool that does appear to support it is Thonny!

**Comment by elmot** [comment]
>For now I have Pycharm 2024.1

Seems to be fixed after 1.44 release. 

>Plus only one file is pushed to the device.. It should be possible to push multiple files.. 
It is possible. Jusk make a micropython run configuration pointing at a folder, and the whole subtree is uploaded when you start it.

I close the discussion for now, if something goes wrong, please file an issue
